pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -46,16 +47,32 @@ const (
 
 const loadBalancerProvisioningStatusActive = "ACTIVE"
 
+// We wrap the net.LookupHost function in a variable to allow overriding it in unit tests.
```

Comment:
Eww, but I don't personally know of a cleaner way to do this. I'll let it pass, but if anybody else knows better... @pierreprinetti maybe?

Reply:
The alternative that pops to mind is dependency injection: If `ReconcileLoadBalancer` can accept a resolver as an additional argument, I'd inject the resolver from the controller. This `lookupHost` woud then be defined inside the controller, rather than being a global var (which can be dangerous). However, my solution doesn't work if you want to unit-test the controller this way.

If that's what you're looking for:
```Go
type Resolver func(string) ([]string, error)

func (s *Service) ReconcileLoadBalancer(openStackCluster *infrav1.OpenStackCluster, clusterName string, apiServerPort int, lookupHost Resolver) (bool, error) {
```
and in `openstackcluster_controller.go`:
```Go
		resolver := func(host string) ([]string, error) {
			ctx, cancel := context.WithTimeout(context.TODO(), time.Second)
			defer cancel()
			return net.DefaultResolver.LookupHost(ctx, host)
		}
		terminalFailure, err := loadBalancerService.ReconcileLoadBalancer(openStackCluster, clusterName, apiServerPort, resolver)

```

However, my suggestion is not really a game changer.

Reply:
Thanks Pierre for your analysis. I think the PR is then ready.